src/rtc_monitor.ts

```diff
@@ -203,18 +191,9 @@ export class RTCMonitor extends EventEmitter {
             }
         });
 
-        if (!candidate) {
-            this.logger.logDebug('RTCMonitor: no valid candidate was found');
-            return;
-        }
-
-        // Step 1: get transport latency from the in-use candidate pair stats, if present.
-        let transportLatency;
-
-        // currentRoundTripTime could be missing in the original report (e.g. on Firefox) and implicitly coverted to NaN.
-        if (!isNaN(candidate.currentRoundTripTime)) {
-            transportLatency = (candidate.currentRoundTripTime * 1000) / 2;
-        }
+        // Step 1: get transport round-trip time from the peer.
+        // This is calculated through ping/pong messages on the data channel.
+        const transportRTT = this.peer.getRTT();
```

Comment:
This will block?

Reply:
@cpoile It's a simpler getter to read `RTCPeer.rtt`, so I'd say no :)

https://github.com/mattermost/calls-common/blob/8d2b13bd2f10847a4be461dd4225fef2ade06ab9/src/rtc_peer.ts#L121-L126